compare-evidence-strings/compare.sh

```diff
@@ -131,19 +121,36 @@ echo "  Diff for evidence strings with *unique* association fields"
 cut -f2 08.common > 10.common.old & cut -f3 08.common > 10.common.new & wait
 compute_git_diff 10.common.old 10.common.new > 09.unique-diff
 
-echo "  Extract functional consequences for all changed evidence strings"
-extract_functional_consequences < 10.common.old > 11.consequences.old \
-  & extract_functional_consequences < 10.common.new > 11.consequences.new \
-  & wait
 
-echo "  Identify consequence type transitions"
-paste 11.consequences.old 11.consequences.new \
-  | awk -F$'\t' '$1 != $2' \
-  | sort > 12.consequences-transitions
 
-echo "  Compute frequencies of consequence type transitions"
-uniq -c 12.consequences-transitions | sort -k1,1rn > 99.consequences-transition-frequency
+########################################################################################################################
+echo "Identify evidence strings with changes in phenotype, datatype, or consequence type"
+
+paste <(cut -d '|' -f 1,2,4,5,6 02.fields.old) <(cut -d '|' -f 3 02.fields.old) | sort -k1,1 > 03a.phenotype.old
+paste <(cut -d '|' -f 1,2,4,5,6 02.fields.new) <(cut -d '|' -f 3 02.fields.new) | sort -k1,1 > 03a.phenotype.new
+
+paste <(cut -d '|' -f 1,2,3,5,6 02.fields.old) <(cut -d '|' -f 4 02.fields.old) | sort -k1,1 > 03a.datatype.old
+paste <(cut -d '|' -f 1,2,3,5,6 02.fields.new) <(cut -d '|' -f 4 02.fields.new) | sort -k1,1 > 03a.datatype.new
 
+paste <(cut -d '|' -f 1,2,3,4,6 02.fields.old) <(cut -d '|' -f 5 02.fields.old) | sort -k1,1 > 03a.consequence.old
+paste <(cut -d '|' -f 1,2,3,4,6 02.fields.new) <(cut -d '|' -f 5 02.fields.new) | sort -k1,1 > 03a.consequence.new
```

Comment:
My personal preference here would be `awk` to avoid reading the same file twice.
```awk
awk 'BEGIN{FS="|"; OFS="|"} function mvcol(col) { tmp=$col; for (i=col; i<NF; i++) {$i = $(i+1)}  NF-- } { mvcol(3); print $0"\t"tmp }'
``` 
This could even be refactor as one of your function

Reply:
This is much nicer, thank you!  Someday I need to sit down and learn awk properly...